### ISSUE TITLE
Investigate WindowsDesktop dependencies on System.Security.Permissions so it can be safely omitted

### ISSUE THREAD

**Issue by dagood** [dotnet/windowsdesktop#920]
While trying to remove `System.Security.Permissions` for https://github.com/dotnet/core-setup/issues/7290, some closure validation errors came up:

```
Assembly 'System.Xaml' is missing dependency 'System.Security.Permissions'
Assembly 'System.Configuration.ConfigurationManager' is missing dependency 'System.Security.Permissions'
Assembly 'System.Security.Cryptography.Xml' is missing dependency 'System.Security.Permissions'
Assembly 'WindowsBase' is missing dependency 'System.Security.Permissions'
Assembly 'System.Windows.Forms.Design.Editors' is missing dependency 'System.Security.Permissions'
Assembly 'System.Drawing' is missing dependency 'System.Security.Permissions'
Assembly 'System.DirectoryServices' is missing dependency 'System.Security.Permissions'
```

Some may be caused by having stale dependencies (https://github.com/dotnet/core-setup/issues/7343), but not all (https://github.com/dotnet/core-setup/pull/7324#discussion_r306436855):

> That's not the only cause for this. Some of these other dependencies are concerning. I suspect we'll need to suppress this, since even once everything is fixed WindowsBase will have typeforwards to System.Security.Permission, but we need to examine all the references before suppressing to find out if there is other work to do to safely omit this.

Once we're sure this is safe (not sure if Core-Setup should be tracking that in particular), we can remove the dependency on `System.Security.Permissions` and add:

```xml
    <IgnoredReference Condition="'$(PackageTargetRuntime)' == ''" Include="System.Security.Permissions" />
```

/cc @ericstj @vatsan-madhavan

**Comment by RussKie** [comment]
Is this still a thing?

**Comment by ericstj** [comment]
Yes, please remove System.Security.Permissions from WindowsDesktop.

**Comment by vatsan-madhavan** [comment]
WPF continues to rely on/use `XamlAccessLevel` and `XamlLoadPermission` because these were exposed in .NET 4.x public contracts and in turn .NET Core 3.x needed to maintain the same contracts as well. 

At some point, WPF would have to break those contracts and remove those S.S.P dependencies - and only then this work can be completed.

**Comment by ericstj** [comment]
We could push types into System.Security.Permissions then leave a dangling typeforward.  It's not all that unlike mscorlib.dll with dangling type-forwards only WPF also happens to put type-defs in WindowsBase.  Would this impact WPF runtime?

**Comment by vatsan-madhavan** [comment]
WPF does more than passively expose these types; it uses `XamlAccessLevel` and `XamlLoadPermission` in runtime (esp. the former)

https://github.com/dotnet/wpf/search?q=xamlaccesslevel

For e.g., `XamlObjectWriter` continues to respect `XamlAccessLevel` 

[src/Microsoft.DotNet.Wpf/test/DRT/DrtXaml/DrtXaml/Tests/AdvXamlFeatureTests.cs#L1046-L1053](https://github.com/dotnet/wpf/blob/826e2df1774c8beaf71d757d776a20b714b6dd5e/src/Microsoft.DotNet.Wpf/test/DRT/DrtXaml/DrtXaml/Tests/AdvXamlFeatureTests.cs#L1046-L1053)
```c#
            // Set local assembly on the XOW also (to access internal Ctors)
            XamlObjectWriterSettings xowSettings = new XamlObjectWriterSettings();
            xowSettings.AccessLevel = System.Xaml.Permissions.XamlAccessLevel.AssemblyAccessTo(localAssembly);


            XamlObjectWriter objWriter = new XamlObjectWriter(xamlReader.SchemaContext, xowSettings);
            XamlServices.Transform(xamlReader, objWriter);
            object o = objWriter.Result;
            Test.Elements.Object10 root = (Test.Elements.Object10)o;
```

Ultimately, this would likely require some refactoring of [`DynamincMethodRuntime`](https://github.com/dotnet/wpf/blob/826e2df1774c8beaf71d757d776a20b714b6dd5e/src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Runtime/DynamicMethodRuntime.cs#L125-L137) and its antecedents:

```c#
        internal DynamicMethodRuntime(XamlRuntimeSettings settings, XamlSchemaContext schemaContext,
            XamlAccessLevel accessLevel)
            : base(settings, true /*isWriter*/)
        {
            Debug.Assert(schemaContext != null);
            Debug.Assert(accessLevel != null);
            _schemaContext = schemaContext;
            _localAssembly = Assembly.Load(accessLevel.AssemblyAccessToAssemblyName);
            if (accessLevel.PrivateAccessToTypeName != null)
            {
                _localType = _localAssembly.GetType(accessLevel.PrivateAccessToTypeName, true /*throwOnError*/);
            }
        }
```

**Comment by dreddy-work** [comment]
adding @pchaurasia14 / @singhashish-wpf to follow up on this.
